### DISCUSSION TITLE
Unpredictable behaviour

### DISCUSSION THREAD

**Discussion by DRBAUS** [RomanKubiak/ctrlr#216] [Q&A]
Hi there, 

I have been busy still with the Panel for the Ensoniq Mirage running the alternative Soundprocess OS. It's coming along quiet well but I am running into some unpredictable behaviour so I  will post it here.

Question 1:
How do you work in standalone mode with a soundmodule and an external keyboard?
I have an M-Audio MIDISPORT 4x4 hooked up. Port 1 in/out is the Mirage and this works well. Data is send handshake style. Top!
Now, on port 2 in I have the Access Virus kb connected for playing. In the MIDI Settings in Ctrlr I have set the MIDI Device to port 2. 

The issue is that, for example last night, I have been playing for hours sending notes from the Virus kb to the Mirage, no problems at all. Then...the next day....bam...the Mirage chokes up on the incoming MIDI notes right away. 

Question 2:
I have a Slider set to respond to MIDI CC 1 (Modwheel). Again the whole evening yesterday it was working perfectly from the Virus kb. Today the Modulator is not responding to incoming Modwheel information at all.

What is it that I am not seeing here?

**Comment by dobo365** [comment]
Can you make simple drawing of your connections?
Please also add screenshots of your Midi menu settings incl. Midi Thru.
Normally, when you have a sound module (as the Behringer ones), you need to:
- set the Input from your controller keyboard
- set the Output to your sound module
- DO NOT set Controller (don't know the reason but I know it is giving issues)
- set Midi Thru : enable Input Device to Output Device to pass notes from the controller keyboard to the module through the panel

Of course if possible, you can use your controller to change buttons in the panel. This will work with the setup I described

**Reply by DRBAUS** [reply]
Thanks Goodweather.

Okay, I understand the connections like you describe. In this way there is no 2-way communication between the Panel and the Mirage but that is actually okay. It works, I can play from the controller keyboard to MIDI Input and the Panel will Output to the Mirage. 

Then the second issue remains a mystery to me. The Slider simply is not responding to Modwheel while I can see the input and output in the Ctrlr Midi Monitor. 

I have tried it with 2 different interfaces as well (M-Audio and Steinberg). Also tried it from within Cubase. M-Audio does not work multi-client tasks, the Steinberg UR242 does but still the Slider does not respond to the Modwheel commands (or Breath Control)

Is it an option to write a script for this as "when panel receives MIDI" or would that slow things down?

**Reply by DRBAUS** [reply]
I think I am on to something with regards to my last question. I found some code on the old forum so let me try that out and get back to you.

**Reply by DRBAUS** [reply]
Alright....I took that hurdle! It's working.

Thanks.
